### PR TITLE
Add timeout for web_channel in trial_runner

### PR DESCRIPTION
**Pull request by SparkSnail**

### CODE REVIEW

File: tools/nni_trial_tool/web_channel.py [hunk 1]
```diff
@@ -16,19 +16,23 @@ def __init__(self, args):
...
-        client = self._event_loop.run_until_complete(connect)
-        self.client = client
+        try:
+            connect = asyncio.wait_for(websockets.connect(url), self.timeout)
+            self._event_loop = asyncio.get_event_loop()
+            client = self._event_loop.run_until_complete(connect)
+            self.client = client
+            nni_log(LogType.Info, 'WebChannel: connected with info %s' % url)
+        except asyncio.TimeoutError:
+            nni_log(LogType.Error, 'WebChannel: connect %s failed!' % url)
```
Comment (squirrelsc) [inline tools/nni_trial_tool/web_channel.py:34]: change the error message like below. It's more precise, and guide user to fix it.

WebChannel: connect to %s timeout! Please make sure NNIManagerIP configured correclty, and accessable.

File: tools/nni_trial_tool/web_channel.py [hunk 2]
```diff
@@ -16,19 +16,23 @@ def __init__(self, args):
...
-        nni_log(LogType.Info, 'WebChannel: connected with info %s' % url)
-
-        connect = websockets.connect(url)
-        self._event_loop = asyncio.get_event_loop()
-        client = self._event_loop.run_until_complete(connect)
-        self.client = client
+        try:
+            connect = asyncio.wait_for(websockets.connect(url), self.timeout)
+            self._event_loop = asyncio.get_event_loop()
+            client = self._event_loop.run_until_complete(connect)
```
Comment (squirrelsc) [inline tools/nni_trial_tool/web_channel.py:30]: Is it possible to hang at this line in some cases? Is a timeout needed for event loop?

File: tools/nni_trial_tool/web_channel.py (hunk 1 above)
Comment (SparkSnail) [reply tools/nni_trial_tool/web_channel.py]: fixed.

File: tools/nni_trial_tool/web_channel.py (hunk 2 above)
Comment (SparkSnail) [reply tools/nni_trial_tool/web_channel.py:30]: This self._event_loop.run_until_complete(connect) is for connect method, and connect has already handled timeout.

File: tools/nni_trial_tool/web_channel.py [hunk 3]
```diff
@@ -16,19 +16,23 @@ def __init__(self, args):
...
+            self.client = client
+            nni_log(LogType.Info, 'WebChannel: connected with info %s' % url)
+        except asyncio.TimeoutError:
+            nni_log(LogType.Error, 'connect to %s timeout! Please make sure NNIManagerIP configured correclty, and accessable.' % url)
```
Comment (chicm-ms) [inline tools/nni_trial_tool/web_channel.py:34]: typo: `correclty`

Comment (SparkSnail) [reply tools/nni_trial_tool/web_channel.py]: fixed

**Pull request merged**